circuit/VenmoMMV1_regex.circom

```diff
@@ -0,0 +1,361 @@
+
+pragma circom 2.0.3;
+
+include "../zk-email-verify-circuits/regex_helpers.circom";
+
+template VenmoMMV1Regex (msg_bytes) {
+    signal input msg[msg_bytes];
+    signal output out;
+
+    var num_bytes = msg_bytes;
+    signal in[num_bytes];
+    for (var i = 0; i < msg_bytes; i++) {
+        in[i] <== msg[i];
+    }
+	
+	component eq[41][num_bytes];
+	component lt[4][num_bytes];
+	component and[45][num_bytes];
+	component multi_or[2][num_bytes];
+	signal states[num_bytes+1][42];
+	
+	for (var i = 0; i < num_bytes; i++) {
+		states[i][0] <== 1;
+	}
+	for (var i = 1; i < 42; i++) {
+		states[0][i] <== 0;
+	}
+	
+	for (var i = 0; i < num_bytes; i++) {
+		eq[0][i] = IsEqual();
+		eq[0][i].in[0] <== in[i];
+		eq[0][i].in[1] <== 47;
+		and[0][i] = AND();
+		and[0][i].a <== states[i][41];
+		and[0][i].b <== eq[0][i].out;
+		states[i+1][1] <== and[0][i].out;
+		eq[1][i] = IsEqual();
+		eq[1][i].in[0] <== in[i];
+		eq[1][i].in[1] <== 99;
+		and[1][i] = AND();
+		and[1][i].a <== states[i][1];
+		and[1][i].b <== eq[1][i].out;
+		states[i+1][2] <== and[1][i].out;
+		eq[2][i] = IsEqual();
+		eq[2][i].in[0] <== in[i];
+		eq[2][i].in[1] <== 111;
+		and[2][i] = AND();
+		and[2][i].a <== states[i][2];
+		and[2][i].b <== eq[2][i].out;
+		states[i+1][3] <== and[2][i].out;
+		eq[3][i] = IsEqual();
+		eq[3][i].in[0] <== in[i];
+		eq[3][i].in[1] <== 100;
+		and[3][i] = AND();
+		and[3][i].a <== states[i][3];
+		and[3][i].b <== eq[3][i].out;
+		states[i+1][4] <== and[3][i].out;
+		eq[4][i] = IsEqual();
+		eq[4][i].in[0] <== in[i];
+		eq[4][i].in[1] <== 101;
```

Comment:
Regex parsing seems wrong. [47, 99,111,100,101] is `/code` so its not starting at the beginning of the preselector string